File: Level1/34.크레인인형뽑기게임/34.크레인인형뽑기게임.cpp
#include <string>
#include <vector>
#include <stack>
using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    stack<int> stk;
    for(auto element:moves){
        element = element-1;
        for(int i = 0 ; i<board.size() ; i++){
            if(board[i][element]>0){
                if(!stk.empty() && stk.top()==board[i][element]){
                    stk.pop();
                    board[i][element] = 0;
                    answer += 2;
                }else{
                    stk.push(board[i][element]);
                    board[i][element] = 0;
                }
                break;
            }
        }
    }
    return answer;
}

File: Level1/34.크레인인형뽑기게임/34.크레인인형뽑기게임.java
import java.util.Stack;
class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stk = new Stack<Integer>();
        for(int i = 0 ; i<moves.length ; i++){
            int element = moves[i]-1;
            for(int j = 0 ; j<board.length ; j++){
                if(board[j][element]>0){
                    if(!stk.empty() && stk.peek()==board[j][element]){
                        stk.pop();
                        board[j][element] = 0;
                        answer += 2;
                    }else{
                        stk.push(board[j][element]);
                        board[j][element] = 0;
                    }
                    break;
                }
            }
        }
        return answer;
    }
}